src/bit-systems/video-texture.ts

```diff
@@ -131,43 +177,33 @@ export function videoTextureSystem(world: HubsWorld) {
     }
   });
 
-  enteredVideoTextureTargetsQuery(world).forEach(function (eid) {
-    const srcData = sourceDataMap.get(VideoTextureTarget.source[eid]);
-    const targetData: TargetData = { originalMap: null, originalEmissiveMap: null };
-    targetDataMap.set(eid, targetData);
-
-    if (!srcData) {
-      console.error("video-texture-target unable to find source");
-      return;
-    }
-
-    const mat = world.eid2mat.get(eid)! as MeshStandardMaterial;
-    if (VideoTextureTarget.flags[eid] & VIDEO_TEXTURE_TARGET_FLAGS.TARGET_BASE_MAP) {
-      targetData.originalMap = mat.map;
-      mat.map = srcData.renderTarget.texture;
-    }
-    if (VideoTextureTarget.flags[eid] & VIDEO_TEXTURE_TARGET_FLAGS.TARGET_EMISSIVE_MAP) {
-      targetData.originalEmissiveMap = mat.emissiveMap;
-      mat.emissiveMap = srcData.renderTarget.texture;
-    }
-  });
   exitedVideoTextureTargetsQuery(world).forEach(function (eid) {
-    const targetData = targetDataMap.get(VideoTextureTarget.source[eid])!;
-    const mat = world.eid2mat.get(eid)! as MeshStandardMaterial;
-    if (VideoTextureTarget.flags[eid] & VIDEO_TEXTURE_TARGET_FLAGS.TARGET_BASE_MAP) {
-      mat.map = targetData.originalMap;
-    }
-    if (VideoTextureTarget.flags[eid] & VIDEO_TEXTURE_TARGET_FLAGS.TARGET_EMISSIVE_MAP) {
-      mat.emissiveMap = targetData.originalMap;
-    }
+    const isBound = targetDataMap.has(eid);
+    if (isBound && entityExists(world, eid)) unbindMaterial(world, eid);
     targetDataMap.delete(eid);
   });
+  videoTextureTargetQuery(world).forEach(function (eid) {
+    const source = VideoTextureTarget.source[eid];
+    const isBound = targetDataMap.has(eid);
+    if (isBound) {
+      if (!source || !entityExists(world, source)) {
+        unbindMaterial(world, eid);
+        VideoTextureTarget.source[eid] = 0;
+      } else if (source !== targetDataMap.get(eid)!.boundTo) {
+        unbindMaterial(world, eid);
+        bindMaterial(world, eid);
+      }
+    } else if (source && entityExists(world, source)) {
+      bindMaterial(world, eid);
+    }
+  });
```

Comment:
What happens if the `source` entity goes away, and this `VideoTextureTarget.source[eid]` pointed to it? Do we need to check for that and set `VideoTextureTarget.source[eid] = 0`, is this handled already, or is this not possible / unnecessary?

Reply:
This should be handled just above `if (!source || !entityExists(world, source)) {`